rules/dotnet-upgrade-for-complex-projects/rule.md

```diff
@@ -1,169 +1,203 @@
+
+Install the tool using 
+
+```bash
+dotnet tool install -g try-convert
+```
+    
+Upgrade your web projects using
+
+```bash
+try-convert --keep-current-tfms --force-web-conversion
+```
+
+and your other projects using
+
+```bash
+try-convert --keep-current-tfms
+```
+
+### Change all your projects to be able to target multiple Target framework monikers (TFM)
+
+In all your project files change the `TargetFramework` to `TargetFrameworks`. You want to do  this early on to enable a smoother flow later to not need unload and reload projects or have to close and reopen Visual Studio.
+
+What this will allow you to do is add your target framework and compile the code. This will allow you to see what code is not compatible with the new framework and fix those issues while still developing/deploying your project in the current target framework. 
+
+```csharp
+<TargetFrameworks>net472;net8.0</TargetFrameworks>
+```
+
+![Figure: Git changes for targeting to multiple target frameworks](target-to-multiple-TFMs.png)
+
+# Upgrading
+
+At this point, ensure your project can target both the .NET Framework and the new target .NET. Some of the projects might not support both platforms right away and you can follow these steps to fix the issues and have a better understanding of how much work it might lies ahead.
+
+1. Add the target framework to your project
+2. Compile to see what breaks
+3. Fix what is easy to fix
+   1. Remember to commit after each fix to help your reviewers 😉
+4. Anything that is not easy to fix, create a PBI with details of the issue
+   1. This allows another developer on your team to work on that PBI independently
+5. If you have a project that is able to compile at this point you can leave the new TFM in your project and continue to the next project
+   1. If not, you can remove the new TFM and continue to the next project
+   2. Repeat these steps once the PBIs have been completed related to this project
+
+Outlined below are rules designed to assist in the project upgrade process during migration. Please note that the applicability of certain rules may vary based on individual project requirements.
+* [Do you know how to migrate from System.Web to modern alternatives?](https://www.ssw.com.au/rules/migrate-from-system-web-to-modern-alternatives/)
+* [Do you know how to migrate from EDMX to EF Core?](https://www.ssw.com.au/rules/migrate-from-edmx-to-ef-core/)
+
+# Web application
+
+There are several ways to migrate project from ASP.NET to ASP.NET Core. We strongly recommend using the Strangler Fig pattern to incrementally migrate your project with [YARP](https://microsoft.github.io/reverse-proxy/).
+
+### Create side-by-side incremental project with [.NET Upgrade Assistant](https://dotnet.microsoft.com/en-us/platform/upgrade-assistant)

```

Comment:
Now that we have the other Upgrade Assistant section below, do we still require this section?

Reply:
As discussed earlier today, this is one of the sections that would benefit of some more details and context.
It owuld be good to have a few steps on how to create the side-by-side, and then some explenation what to do next once that is done eg. setup the routes on your frontend to point to the proxy
and configure the proxy to add a catchall to send everything to the original project.

And then it continues with my edits.